x/pairing/keeper/msg_server_unfreeze.go

```diff
@@ -24,6 +24,17 @@ func (k msgServer) UnfreezeProvider(goCtx context.Context, msg *types.MsgUnfreez
 			return nil, utils.LavaFormatWarning("Unfreeze_cant_get_stake_entry", types.FreezeStakeEntryNotFoundError, []utils.Attribute{{Key: "chainID", Value: chainId}, {Key: "providerAddress", Value: msg.GetCreator()}}...)
 		}
 
+		minStake := k.Keeper.dualstakingKeeper.GetMinStake(ctx, chainId)
```

Comment:
GetMinStake is from the spec keeper so iinstead of calling dualstaking->spec
call directly the spec keeper

Reply:
Yes, but there's a small logic there: 
```
func (k Keeper) GetMinStake(ctx sdk.Context, chainID string) sdk.Coin {
	spec, found := k.specKeeper.GetSpec(ctx, chainID)
	if !found {
		utils.LavaFormatError("critical: failed to get spec for chainID",
			fmt.Errorf("unknown chainID"),
			utils.Attribute{Key: "chainID", Value: chainID},
		)
	}

	return spec.MinStakeProvider
}
```

Reply:
can you move it to the spec keeper please :)

Reply:
Done in 4b4a61c5e9f7b3e0a108bd207caa91da823783b5